src/rx-query.ts

```diff
@@ -634,6 +740,7 @@ function __ensureEqual<RxDocType>(rxQuery: RxQueryBase<RxDocType>): Promise<bool
                 if (newCount !== previousCount) {
                     ret = true; // true because results changed
                     rxQuery._setResultData(newCount as any);
+                    await updatePersistentQueryCache(rxQuery);
```

Comment:
we have to await here? cannot just update pers query cache async?

Reply:
This is for the sake of consistency. We wait till the result of `count` is stored in `Cache`, before we return the result to user code. Otherwise, we might show a value that we never persist, and the next query with the same ID could increment from a different value in case of any change events.

Reply:
OK

Reply:
i wonder if we should just skip persistence for counts<X. i guess those counts could still be slow 🤔  i guess we can just choose where to persist vs not. for example, i might suggest we only persist counts of non-built-in queries